### PR TITLE
fix: edge case in app.isInApplicationsFolder()

### PR DESCRIPTION
**Pull request by KishanBagaria**
#### Description of Change
This fixes an edge case in `app.isInApplicationsFolder()` which returns false when an app is launched from Terminal like this:
```
/applications/App.app/Contents/MacOS/App
```
while this works fine:
```
/Applications/App.app/Contents/MacOS/App
```

(Altho this seems unlikely to affect things much, we've received at least one user report of `app.isInApplicationsFolder()` returning false incorrectly when the app is launched from Finder and this has some likelihood of fixing that.)

#### Checklist

- [ ] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [ ] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an edge case in `app.isInApplicationsFolder()` which would return false incorrectly in some cases.

### CODE REVIEW

**Review by ckerr — DISMISSED** [review]

**Review by miniak — CHANGES_REQUESTED** [review]
the filesystem can be formatted as case-sensitive

**Comment by miniak** [comment]
https://stackoverflow.com/questions/4706215/mac-os-x-how-to-determine-if-filesystem-is-case-sensitive

**Comment by KishanBagaria** [comment]
@miniak can you say more about how the change should be implemented?

**Comment by MarshallOfSound** [comment]
You need to normalize the bundle path (effectively call realpath on it)

**Comment by KishanBagaria** [comment]
Good idea, done.

**Review by codebytere — APPROVED** [review]

**Review by ckerr — APPROVED** [review]

**Review by miniak — APPROVED** [review]

File: shell/browser/ui/cocoa/electron_bundle_mover.mm [hunk 1]
```diff
@@ -181,19 +181,27 @@
 bool ElectronBundleMover::IsCurrentAppInApplicationsFolder() {
   return IsInApplicationsFolder([[NSBundle mainBundle] bundlePath]);
 }
+  
+NSString* resolvePath(NSString* path) {
+  NSString *standardizedPath = [path stringByStandardizingPath];
+  char resolved[PATH_MAX];
+  if (realpath([standardizedPath UTF8String], resolved) == NULL) return path;
+  return [NSString stringWithCString:resolved encoding:NSUTF8StringEncoding];
```
Comment (miniak) [inline shell/browser/ui/cocoa/electron_bundle_mover.mm:189]: you can use `[NSString stringWithUTF8String:resolved]` or just @(resolved)`

**Review by miniak — CHANGES_REQUESTED** [review]

File: shell/browser/ui/cocoa/electron_bundle_mover.mm (hunk 1 above)
Comment (KishanBagaria) [inline shell/browser/ui/cocoa/electron_bundle_mover.mm:189]:
```suggestion
  return @(resolved);
```

Comment (KishanBagaria) [reply shell/browser/ui/cocoa/electron_bundle_mover.mm]: done, thx

**Review by miniak — APPROVED** [review]

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**